src/Git/Commit.php

```diff
@@ -0,0 +1,60 @@
+<?php
```

Comment:
This gives you way more possibilities to make further refactoring. You could introduce `Tag` class and get rid of git gateway dependencies in your releases. Then you could move some logic inside of this object (if you see getters somewhere). This will make your "release class" to be isolated unit which could be tested independently.

Also this simplifies GitGateway interface, reduces its responsibilities. 

And for list of 20 commits you are now do only one exec instead of 20 process spawn.